### ISSUE TITLE
failing test and upgrade to to mio 1.0?

### ISSUE THREAD

**Issue by aconz2** [samuelbrian/mio-pidfd#5]
This is a very useful crate for integrating with mio, thank you!

I was getting a weird error about missing impl for Source and I think the reason came from mio-pidfd depending on mio 0.8 and my app was using 1.0.2 (new to Rust but I think that's right). Anyways was testing if the upgrade was okay and noticed two tests were failing (at least on kernel 6.10).

Maybe this changed but it looks like there are always two read events that happen when a process terminates, the first has read_closed=false and the second has read_closed=true. The single_process test fails asserting that the events are empty at the second read (which makes sense to me but doesn't seem to be the case).

```
---- tests::single_process stdout ----
thread 'tests::single_process' panicked at src/lib.rs:283:9:
assertion failed: events.is_empty()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

with this patch:

```patch
diff --git a/src/lib.rs b/src/lib.rs
index 89d4a9c..b178bdf 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -271,6 +271,7 @@ mod tests {
         // child should exit with an event
         poll.poll(&mut events, Some(TIMEOUT2)).unwrap();
         assert!(!events.is_empty());
+        println!("{events:?}");
         match child.try_wait() {
             Ok(Some(_)) => {}
             Ok(None) => panic!("poll event occured but child has not exited"),
@@ -280,7 +281,12 @@ mod tests {
         // child should not cause event again after exit
         poll.poll(&mut events, Some(Duration::from_nanos(1)))
             .unwrap();
-        assert!(events.is_empty());
+        //assert!(events.is_empty());
+        println!("{events:?}");
+
+        poll.poll(&mut events, Some(Duration::from_nanos(1)))
+            .unwrap();
+        println!("{events:?}");
     }
 
     #[test]

```

this is what I see:

```
---- tests::single_process stdout ----
[Event { token: Token(2), readable: true, writable: false, error: false, read_closed: false, write_closed: false, priority: false, aio: false, lio: false }]
[Event { token: Token(2), readable: true, writable: false, error: false, read_closed: true, write_closed: true, priority: false, aio: false, lio: false }]
[]
```

and the corresponding strace output:

```
# strace -e status=successful -e epoll_wait -f cargo test single_process
[pid 1215164] epoll_wait(3, [], 1024, 1000) = 0
[pid 1215165] +++ exited with 0 +++
[pid 1215164] epoll_wait(3, [{events=EPOLLIN, data={u32=2, u64=2}}], 1024, 2000) = 1
[pid 1215164] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=1215165, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
[pid 1215164] epoll_wait(3, [{events=EPOLLIN|EPOLLHUP, data={u32=2, u64=2}}], 1024, 1) = 1
[pid 1215164] epoll_wait(3, [], 1024, 1) = 0
```

I'm having a hard time tracking down a definitive source on whether this is expected or not, maybe you know? Thanks!

**Comment by aconz2** [comment]
Cool thanks for the info, that matches up with the behavior described in the 6.9 release notes. I just did a quick test and if you only `epoll_wait` after the child is reaped, you only get a single event with `EPOLLIN|EPOLLHUP` which makes sense. For my use it is sufficient to just receive the first event so this isn't a stopper issue, but good to know. Thanks again

**Comment by samuelbrian** [comment]
I've updated the failing tests to tolerate both epoll behaviours now. 
I also bumped the mio dependency to 1.0, found no problems with the tests, so published the changes to this crate as 0.4.0.
Hopefully that helps you out.

**Comment by aconz2** [comment]
Awesome just pulled it, much appreciated!
